### PR TITLE
Refactor webhook framework to allow webhooks define their flags

### PR DESCRIPTION
**Pull request by alpeb**
Pulled out of `launcher.go` the flag parsing logic and moved it into the `Main` methods of the webhooks (under `controller/cmd/proxy.injector/main.go` and `controller/cmd/sp-validator/main.go`), so that individual webhooks themselves can define the flags they want to use.

Also no longer require that webhooks have cluster-wide access.

Finally, renamed the type `webhook.handlerFunc` to `webhook.Handler` so it can be exported. This will be used in the upcoming jaeger webhook.

### CODE REVIEW

File: controller/cmd/proxy-injector/main.go [hunk 1]
```diff
@@ -2,21 +2,30 @@ package proxyinjector
...
+	"github.com/linkerd/linkerd2/pkg/flags"
 )
 
 // Main executes the proxy-injector subcommand
 func Main(args []string) {
+	cmd := flag.NewFlagSet("proxy-injector", flag.ExitOnError)
+	metricsAddr := cmd.String("metrics-addr", fmt.Sprintf(":%d", 9995), "address to serve scrapable metrics on")
+	addr := cmd.String("addr", ":8443", "address to serve on")
+	kubeconfig := cmd.String("kubeconfig", "", "path to kubeconfig")
+	flags.ConfigureAndParse(cmd, args)
```
Comment (alpeb) [inline controller/cmd/proxy-injector/main.go:20]: The upcoming jaeger webhook will define a couple of extra flags here. We'll pass them to the handler function below (`Injector.Inject`)  in this case.

File: controller/webhook/launcher.go [hunk 2]
```diff
@@ -2,36 +2,33 @@ package webhook
...
+	metricsAddr string,
+	addr string,
+	kubeconfig string,
+) {
 	stop := make(chan os.Signal, 1)
 	defer close(stop)
 	signal.Notify(stop, os.Interrupt, syscall.SIGTERM)
 
-	k8sAPI, err := k8s.InitializeAPI(ctx, *kubeconfig, true, APIResources...)
+	k8sAPI, err := k8s.InitializeAPI(ctx, kubeconfig, false, APIResources...)
```
Comment (alpeb) [inline controller/webhook/launcher.go:31]: Webhooks shouldn't require cluster-wide access.

**Review by adleong — APPROVED** [review]
Nice!

**Review by Pothulapati — APPROVED** [review]
LGTM!

**Pull request merged**